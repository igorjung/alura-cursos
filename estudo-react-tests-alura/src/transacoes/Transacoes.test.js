import React from 'react';
import { render } from '@testing-library/react';

import Transacoes from './Transacoes';

describe('Componente de lista de transações', () => {
  it('O snapshot do component deve permanecer sempre o mesmo', () => {
    const transacoes = [{
      id: 1,
      transacao: 'saque',
      valor: '20.00',
      data: '16/03/2021',
    }, {
      id: 2,
      transacao: 'deposito',
      valor: '15.00',
      data: '17/03/2021',
    }]

    const { container } = render(<Transacoes transacoes={transacoes}/>)
    expect(container.firstChild).toMatchSnapshot();
  })
})