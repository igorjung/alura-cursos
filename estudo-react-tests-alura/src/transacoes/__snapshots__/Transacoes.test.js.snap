// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de lista de transações O snapshot do component deve permanecer sempre o mesmo 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      16/03/2021
    </p>
    <p>
      saque
    </p>
    <p>
      R$ 
      20.00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      17/03/2021
    </p>
    <p>
      deposito
    </p>
    <p>
      R$ 
      15.00
    </p>
  </div>
</div>
`;
